packages/components/src/modal/index.tsx

```diff
@@ -2,7 +2,6 @@
  * External dependencies
  */
 import clsx from 'clsx';
-import type { ForwardedRef, KeyboardEvent, RefObject, UIEvent } from 'react';
```

Comment:
I removed the named imports in favour of using the `React` global namespace because IMO it makes much easier to distinguish between native event types (ie. `KeyboardEvent`) and react synthetic event types (ie. `React.KeyboardEvent`).

Reply:
Could we add an explicit `import type * as React from 'react'` so that we don't use the `React` name without importing it? That also achieves the goal and is easier to understand. For example I don't understand where does the `React` namespace magically appear from 🙂

Reply:
I came across two issues when doing that:

- the code that you propose breaks the `@typescript-eslint/consistent-type-imports` ESLint rule with the error "Type import "React" is used by decorator metadata.". ESLint wants to auto-fix this to `import * as React from 'react'`
- with the auto-fix, the `@typescript-eslint/no-restricted-imports` rule is broken, since ESLint would like us to import it from `@wordpress/element`

I don't think the auto-fix makes sense, since we only want to import the types — but at the same time, I don't understand the first error, and I'm not sure if it's safe to ignore.